### DISCUSSION TITLE
DietPi-Software | Rest Server

### DISCUSSION THREAD

**Discussion by PartTimeDataScientist** [MichaIng/DietPi#5506] [Software Request]
Hi! I'd like to suggest a "Rest Server" (a http server backend for restic-backup) to be added to the Software repository

Original Github description:
_Rest Server is a high performance HTTP server that implements restic's [REST backend API](https://restic.readthedocs.io/en/latest/100_references.html#rest-backend). It provides secure and efficient way to backup data remotely, using [restic](https://github.com/restic/restic) backup client via the [rest: URL](https://restic.readthedocs.io/en/latest/030_preparing_a_new_repo.html#rest-server)._

#### Formal software information
- Title: Rest Server
- Short description: http server backend for restic backup
- Github URL: [Rest-Server on Github](https://github.com/restic/rest-server)
- Main Project / Client: [Restic](https://restic.net) / [Restic on Github] (https://github.com/restic/restic)

#### Are there similar/alternative software titles available with DietPi-Software?
Not that I am aware of. A Borg backup server might be an alternative but that is also not (yet) available

#### What makes your requested software better than the above solutions, if available?
Not applicable

#### How can DietPi make the installation easier or compatible, than following the install instructions or do APT installation, if available?
- Provide an easy means to set the proper runtime options and to create the required .htpasswd file
- Preconfigure a "bastion backup" which allows _append only_ over the network and runs a local instance of restic with sensible defaults for check, prune and forget

#### Can you provide the installation steps that you would suggest DietPi-Software to do?
1.) Download latest suitable release from [Github](https://github.com/restic/rest-server/releases)
2.) Unpack to proper directory
3.) Get user input for manual configuration options where useful (Users, Folders, Append-Only, ...)
*Optional:*
4.) Install local restic client (should be as simple as `apt-get install restic`) for local check, prune, forget in "append-only / bastion" case
5.) Configure local client with for prune/forget with user configurable "sensible defaults"

#### Are you willing to help maintaining the software installation, e.g. in case of needed setup changes due to updates etc.? 
I am planning to set up an instance on a RasPi4 myself so I will likely monitor available updates for security and compatibility. I should be able to adapt a working script to a new release but might need some input if it is getting more complex

**Comment by Joulinar** [comment]
I moved the request die GitHub discussions. Vote for this software implementation with the ⬆️ button on this post.
